ethdb/walk.go

```diff
@@ -93,39 +94,43 @@ func (sc *splitCursor) Next() (key1, key2, key3, val []byte, err error) {
 
 var EndSuffix = []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
 
-func GetModifiedAccounts(db Getter, startTimestamp, endTimestamp uint64) ([]common.Address, error) {
-	keys := make(map[common.Address]struct{})
-	startCode := dbutils.EncodeTimestamp(startTimestamp)
-	if err := db.Walk(dbutils.PlainAccountChangeSetBucket, startCode, 0, func(k, v []byte) (bool, error) {
-		keyTimestamp, _ := dbutils.DecodeTimestamp(k)
+// GetModifiedAccounts returns a list of addresses that were modified in the block range
+func GetModifiedAccounts(tx Tx, startNum, endNum uint64) ([]common.Address, error) {
 
-		if keyTimestamp > endTimestamp {
-			return false, nil
+	changedAddrs := make(map[common.Address]struct{})
+	startCode := dbutils.EncodeTimestamp(startNum)
+
+	c := tx.Cursor(dbutils.PlainAccountChangeSetBucket)
+	for k, v, err := c.Seek(startCode); k != nil; k, v, err = c.Next() {
+		if err != nil {
+			fmt.Println("Error:", err)
```

Comment:
you can do `return nil, fmt.Errorf("iterating over account changeset: %w", err)` to bubble it up